readthedocs/builds/constants.py

```diff
@@ -148,3 +148,19 @@
 MAX_BUILD_COMMAND_SIZE = 1000000  # This keeps us under Azure's upload limit
 
 LOCK_EXPIRE = 60 * 180  # Lock expires in 3 hours
+
+# All artifact types supported by Read the Docs.
+# They match the output directory (`_readthedocs/<artifact type>`)
+ARTIFACT_TYPES = (
+    "html",
+    "json",
+    "htmlzip",
+    "pdf",
+    "epub",
+)
+# Artifacts that are not deleted when uploading to the storage,
+# even if they weren't re-built in the build process.
```

Comment:
```suggestion
# Artifacts that are not deleted when uploading to storage.
# This is because we want to only upload changed files,
# whereas single-file types will need to be deleted & recreated on any change.
```

I think this is the thinking here. Will this be true once we support multiple PDF's? 🤔

Reply:
Hrm, thinking a little more here, I think that both comments (mine and yours) are wrong 😅 

Taking into account that HTML _is mandatory_ for a build, we will always update the HTML output in our storage. Since JSON _is required_ for search we also always want to have it there.

The other formats may be required to delete them when they are _disabled after being enabled_. If the build for version `v3` outputs a PDF and then the user disable the PDF for that version, we have to remove the PDF from the storage. This can't happen for HTML nor JSON because those _are mandatory_.

I complicated myself trying to explain this, but hopefully, I'm being clear enough for you to understand me 😄

Reply:
I'm not 100% following... But it's probably not too important as long as we have a reasonably correct comment.